rust_snuba/src/types.rs

```diff
@@ -183,8 +182,7 @@ impl BytesInsertBatch {
 
 #[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq)]
 pub struct RowData {
-    encoded_rows: Vec<u8>,
-    num_rows: usize,
+    encoded_rows: Vec<Vec<u8>>,
```

Comment:
this is a rather inefficient encoding since you will have one allocation per row. I think it should be possible to let a row go across chunks, or if not, split by `\n` again in the clickhouse strategy. I'm 80% sure either way is more efficient

Reply:
Good point, reverted  this